### PR TITLE
Merge dev

### PR DESCRIPTION
**Pull request by Fraser999**
This brings `dev` into `feat-track-node-2.0` in order to fix a build breakage due to an update to the `casper-types` dep specified in `[patch.crates-io]`.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "casper-client"
 version = "2.0.0" # when updating, also update 'html_root_url' in lib.rs
-authors = ["Marc Brinkmann <marc@casperlabs.io>", "Fraser Hutchison <fraser@casperlabs.io>",]
+authors = ["Marc Brinkmann <marc@casperlabs.io>", "Fraser Hutchison <fraser@casperlabs.io>", "Zachary Showalter <zach@casperlabs.io>"]
```
Comment (bpr) [inline Cargo.toml:4]: Excellent, thanks Fraser!

**Review by zacshowa — APPROVED** [review]
LGTM!

File: lib/cli/parse.rs [hunk 2]
```diff
@@ -376,7 +377,7 @@ macro_rules! check_exactly_one_not_empty {
                 });
             }
         } else {
-            let mut non_empty_fields_with_values = vec![$((stringify!($x), $x)),+]
+            let mut non_empty_fields_with_values = [$((stringify!($x), $x)),+]
```
Comment (bpr) [inline lib/cli/parse.rs:382]: Gotta admit,  I had to look this up. Not just the change, but the original code. So, this is subtle enough to deserve either rewriting into simpler code, or a comment.

I'd say the same about a complex regexp too. Any place where the average reader would do a double take deserves a comment.

**Review by bpr — APPROVED** [review]
Please add a comment where I asked, explaining the unusual (IMO) code. Other than that, LGTM

File: lib/cli/parse.rs (hunk 2 above)
Comment (Fraser999) [reply lib/cli/parse.rs:382]: Done in [157832d](https://github.com/casper-ecosystem/casper-client-rs/pull/101/commits/157832dd2d6f4e4b85b952e8d456388064ec0206).

**Pull request merged**